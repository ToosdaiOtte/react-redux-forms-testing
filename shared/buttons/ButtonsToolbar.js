import ButtonToolbar from 'react-bootstrap/lib/ButtonToolbar';
import Button from 'react-bootstrap/lib/Button';

export default function ButtonsToolbar({handleSubmit, onSubmit, onSubmitAndNavigate, onNavigate, pristine, submitting}) {
    return (
        <ButtonToolbar>
            <Button className="button button--accent" type="button" onClick={handleSubmit(onSubmit)} disabled={pristine || submitting}>
                {pgettext('mortgage-object-evaluation', 'Save Data')}
            </Button>
            <Button className="button button--secondary" type="button" onClick={handleSubmit(onSubmitAndNavigate)} disabled={pristine || submitting}>
                {pgettext('customer-data-prefill', 'Save & Next')}
            </Button>
            <Button className="button button--secondary" type="button" onClick={onNavigate}>
                {pgettext('customer-data-prefill', 'Next')}
            </Button>
        </ButtonToolbar>
    );
}
