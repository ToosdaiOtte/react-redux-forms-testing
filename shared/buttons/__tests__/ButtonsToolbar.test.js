import { shallow } from 'enzyme';
import ButtonsToolbar from '../ButtonsToolbar';

let ButtonsToolbarComponent;
const defaultProps = {
        handleSubmit: (fn) => fn,
        onSubmit: jest.fn(),
        onSubmitAndNavigate: jest.fn(),
        onNavigate: jest.fn()
    },
    ButtonsToolbarWrapper = (props) => {
        return <ButtonsToolbar {...defaultProps} {...props} />;
    };

beforeEach(() => {
    ButtonsToolbarComponent = shallow(
        <ButtonsToolbarWrapper />
    ).dive().children();
});

describe('Render ButtonsToolbarComponent component', () => {
    it('create snapshot for ButtonsToolbarComponent', () => {
        expect(ButtonsToolbarComponent).toMatchSnapshot();
    });

    it('check event on `Save` button', () => {
        const propertySelfOwnedFormButton = ButtonsToolbarComponent.at(0);
        propertySelfOwnedFormButton.simulate('click');
        expect(defaultProps.onSubmit).toHaveBeenCalled();
    });

    it('check event on `Save & continue` button', () => {
        const propertySelfOwnedFormButton = ButtonsToolbarComponent.at(1);
        propertySelfOwnedFormButton.simulate('click');
        expect(defaultProps.onSubmitAndNavigate).toHaveBeenCalled();
    });

    it('check event on `Next` button', () => {
        const propertySelfOwnedFormButton = ButtonsToolbarComponent.at(2);
        propertySelfOwnedFormButton.simulate('click');
        expect(defaultProps.onNavigate).toHaveBeenCalled();
    });
});
