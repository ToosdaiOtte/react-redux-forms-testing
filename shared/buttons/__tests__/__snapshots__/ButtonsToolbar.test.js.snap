// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonsToolbarComponent component create snapshot for ButtonsToolbarComponent 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="button button--accent"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Save Data
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="button button--secondary"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Save & Next
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="button button--secondary"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Next
  </Button>,
]
`;
